midx.c

```diff
@@ -457,8 +459,11 @@ static void add_pack_to_midx(const char *full_path, size_t full_path_len,
 	struct pack_list *packs = (struct pack_list *)data;
 
 	if (ends_with(file_name, ".idx")) {
-		if (packs->m && midx_contains_pack(packs->m, file_name))
+		packs->pack_paths_checked++;
```

Comment:
Perhaps add one `display_progress()` call here, instead of tracking all the `return`s. It would present an off-by-one error, but the time spent in this `if` block is pretty short.